utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -383,6 +420,244 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
+                            }
+                        } else {
+                            trie_value = self.header.null_value;
+                            value = null_value;
+                            have_value = true;
+                        }
+                        c = (c + data_block_length) & !data_mask;
+                    } else {
+                        let mut di: u32 = block + (c & data_mask);
+                        let mut trie_value_2: u32 = if let Some(trv2) = self.data.get(di as usize) {
+                            trv2.into()
+                        } else {
+                            return None;
+                        };
+                        if have_value {
+                            if trie_value_2 != trie_value {
+                                if maybe_filter_value(
+                                    trie_value_2,
+                                    self.header.null_value,
+                                    null_value,
+                                ) != value
+                                {
+                                    return Some(CodePointMapRange {
+                                        start,
+                                        end: c - 1,
+                                        value,
+                                    });
+                                }
+                                trie_value = trie_value_2; // may or may not help
+                            }
+                        } else {
+                            trie_value = trie_value_2;
+                            value = maybe_filter_value(
+                                trie_value_2,
+                                self.header.null_value,
+                                null_value,
+                            );
+                            have_value = true;
+                        }
+
+                        c += 1;
+                        while (c & data_mask) != 0 {
+                            di += 1;
+                            trie_value_2 = if let Some(trv2) = self.data.get(di as usize) {
+                                trv2.into()
+                            } else {
+                                return None;
+                            };
+                            if trie_value_2 != trie_value {
+                                if maybe_filter_value(
+                                    trie_value_2,
+                                    self.header.null_value,
+                                    null_value,
+                                ) != value
+                                {
+                                    return Some(CodePointMapRange {
+                                        start,
+                                        end: c - 1,
+                                        value,
+                                    });
+                                }
+                                trie_value = trie_value_2; // may or may not help
```

Comment:
This comment should either be expanded or removed.

Reply:
(Ditto for the same comment above.)

Reply:
Can you expand the comment, then? What is the purpose of this assignment, and what does it (possibly) help with?

Reply:
Done.

Reply:
@markusicu can you review whether this new text is reasonable?  It was added in `codepointtrie.rs` at L652-661 and duplicated in L690-699.

Reply:
Looks ok; let me describe it in my own words and see if you want to rephrase it.

We have a transition from `trie_value` to `trie_value2`, but we continue the current `range` because both map to the same `value`. The current range might continue yet further with either `trie_value` or `trie_value2` or even something else that also maps to `value`. Functionally, it does not matter which of the trie values we remember, but, on the assumption that often times consecutive characters map to the same trie values, remembering the new one might make it faster to extend this range further (by increasing the chance that the next `trie_value_2 != trie_value` test will be false and not having to call `maybe_filter_value()` as often).

Reply:
I think my description is close enough to be able to merge. I'm also considering that this PR is being waited on by the imminent 0.5 release process, and these source comments are the only part of this PR that hadn't been reviewed.

I'm also adding it to the list of tasks for the immediate followup cleanup PR #1551 that I already got started on, so I'll address any necessary changes here very soon. But for now, I'll go ahead and get the current PR landed. Thanks for all of the help!